Automattic/newspack-plugin
# feat: deprecate test mode and never frequency

### All Submissions:






### Changes proposed in this Pull Request:

Along with https://github.com/Automattic/newspack-popups/pull/390 this simplifies campaign settings in a few ways:

- Removes Test Mode. Recently added Preview and "preview as segment" functionality provides a richer approach and should replace Test Mode.
- Removes the "never" frequency. This setting is redundant as unpublishing is now the preferred approach.

**~Note: this is a breaking change because campaigns with frequencies of "never" or test mode will no longer work. This shouldn't be deployed until a migration strategy has been implemented.~**

**Revised Note: This branch (more specifically the `newspack-popups` one) now contains migration logic.**

The migration approach examines each popup immediately after retrieval from the database. If the frequency is `test` or `never`, the campaign is unpublished. The frequency is set to `daily` for overlays and `always` for inline/above header.
 Theoretically this operation will occur at most once per site. The logic doesn't incur any processing cost, though, since the frequencies that trigger migration are wiped out forever during migration.

Replaces https://github.com/Automattic/newspack-plugin/pull/776, which had grown too out of date with a variety of recent changes. Also replaces https://github.com/Automattic/newspack-popups/pull/366.

### How to test the changes in this Pull Request:

Switch `newspack-popups` to `feat/deprecate-test-mode-two`. 

1. Observe Test mode UI and "Never" frequency have been removed from the editor sidebar and the Newspack Campaigns wizard pop-overs.
2. Verify front end display of campaigns continues to work as expected.

Verify campaign migration:

1. Switch to `master` branches (in both repositories).
2. Create an Overlay campaign in Test mode, published.
3. Create an Inline campaign with frequency Never, published.
4. Create an Inline campaign in Test mode, published.
4. Create an Above Header campaign in Test mode, published.
5. Create at least one inline campaign set to display normally, published.
6. Create one campaign, leave it unpublished.
7. View the site in a logged-in window. You should see the normal inline campaign and all the test mode ones.
8. View the site incognito - you should see only the normal inline campaign.
9. Switch to `feat/deprecate-test-mode-two` (`newspack-plugin` and `newspack-popups`)
10. View the site in a logged-in window. You should see no campaigns.
11. View the site in an incognito window. You should see only the normal inline campaign.
12. Navigate to the Campaigns Wizard. Observe all but the normal/inline one have been unpublished. 
13. Observe Above Header and Inline ones that were unpublished now have frequency Every Page.
14. Observe Overlay now has frequency Once A Day.

### Other information:
